File: xssfinder-java-executor/src/main/java/org/xssfinder/runner/SimpleMethodTraversalStrategy.java
package org.xssfinder.runner;

import org.xssfinder.remote.TUntraversableException;
import org.xssfinder.remote.TraversalMode;

import java.lang.reflect.InvocationTargetException;
import java.lang.reflect.Method;
import java.util.HashMap;
import java.util.Map;

public class SimpleMethodTraversalStrategy implements TraversalStrategy {
    @Override
    public boolean canSatisfyMethod(Method method, TraversalMode traversalMode) {
        return true;
    }

    @Override
    public TraversalResult traverse(Object page, Method method) throws TUntraversableException {
        method.setAccessible(true);
        try {
            Object newPage = method.invoke(page);
            Map<String, String> inputIdsToAttackIds = new HashMap<String, String>();
            return new TraversalResult(newPage, inputIdsToAttackIds);
        } catch (IllegalAccessException e) {
            throw new RuntimeException("Unexpected IllegalAccessException", e);
        } catch (InvocationTargetException e) {
            throw new RuntimeException("Unexpected InvocationTargetException", e);
        } catch (IllegalArgumentException e) {
            throw new TUntraversableException("Cannot invoke method with args");
        }
    }
}

File: xssfinder-php-executor/src/XssFinder/Runner/SimpleMethodTraversalStrategy.php
<?php

namespace XssFinder\Runner;

use ReflectionMethod;
use XssFinder\TraversalMode;
use XssFinder\TUntraversableException;

class SimpleMethodTraversalStrategy implements TraversalStrategy
{
    /**
     * @param ReflectionMethod $method
     * @param int $traversalMode
     * @return boolean
     */
    function canSatisfyMethod(ReflectionMethod $method, $traversalMode)
    {
        return true;
    }

    /**
     * @param mixed $page
     * @param ReflectionMethod $method
     * @return TraversalResult
     *
     * @throws TUntraversableException
     */
    function traverse($page, ReflectionMethod $method)
    {
        if ($method->getNumberOfRequiredParameters() > 0)
        {
            $methodName = $method->getName();
            $numParams = $method->getNumberOfRequiredParameters();
            throw new TUntraversableException(array(
                'message' => "$methodName requires $numParams params - simple method traversal cannot handle this"
            ));
        }
        $newPage = $method->invoke($page);
        return new TraversalResult($newPage, array());
    }
}